File: JavaScript/7js 2/inAscOrder.js
/*
|--------------------------------------------------------------------------
| Are the numbers in order?
|--------------------------------------------------------------------------
|
*/

function inAscOrder(arr) {
   const temp = [...arr].sort((a, b) => a - b);
   return arr.join() == temp.join();
}


function inAscOrder(arr) {
   for (let i = 0; i < arr.length - 1; i++) {
      if (arr[i] > arr[i + 1]) {
         return false;
      }
   }
   return true;
}


function inAscOrder(arr) {
   return arr.every((_, i) => i == 0 || arr[i] > arr[i - 1]);
}


console.log(inAscOrder([1, 2, 4, 7, 19]));
console.log(inAscOrder([1, 2, 3, 4, 5]));
console.log(inAscOrder([1, 6, 10, 18, 2, 4, 20]));
console.log(inAscOrder([9, 8, 7, 6, 5, 4, 3, 2, 1]));

File: Python/7py 2/in_asc_order.py
"""
|--------------------------------------------------------------------------
| Are the numbers in order?
|--------------------------------------------------------------------------
|
"""

def in_asc_order(arr):
   return arr == sorted(arr)


def in_asc_order(arr):
   for i in range(len(arr)-1):
      if arr[i] >= arr[i+1] : 
         return False
   return True


def in_asc_order(a):
   return all(x < y for x, y in zip(a, a[1:]))


print(in_asc_order([1, 2, 3]), True)
print(in_asc_order([1, 3, 2]), False)